src/LanguageServerProtocol/LanguageServerProtocol.fs

```diff
@@ -1665,6 +1680,40 @@ module Types =
         ActiveParameter: int option
     }
 
+    type SignatureHelpTriggerKind =
+    /// manually invoked via command
+    | Invoked = 1
+    /// trigger by a configured trigger character
+    | TriggerCharacter = 2
+    /// triggered by cursor movement or document content changing
+    | ContentChange = 3
+
+    type SignatureHelpContext =
+      {
+        /// action that caused signature help to be triggered
+        TriggerKind: SignatureHelpTriggerKind
+        /// character that caused signature help to be triggered. None when kind is not TriggerCharacter.
+        TriggerCharacter: string option
+        /// true if signature help was already showing when this was triggered
+        IsRetrigger: bool
+        /// the current active SignatureHelp
+        ActiveSignatureHelp: SignatureHelp option
+
+      }
+
+    type SignatureHelpParams =
```

Comment:
filling out new data types for signaturehelp, because we'll likely need them (to get triggering chars and such)